RULES.md

```diff
@@ -229,6 +230,12 @@ Field `parent_station` must be empty when `location_type` is 2
 #### References:
 * [stops.txt specification](http://gtfs.org/reference/static/#stopstxt)
 
+<a name="E043"/>
+
+### E043 - Missing trip edge `arrival_time` and `departure_time`
+
+First and last stop of a trip must define one or both fields
```

Comment:
My interpretation of the spec is that both `arrival_time` and `departure_time` for both the first and last stop_time of a trip *must* be populated. If any of these 4 field values is empty, it's an error.

https://gtfs.org/reference/static#stop_timestxt says:

>An arrival time must be specified for the first and the last stop in a trip.

and

>If there are not separate times for arrival and departure at a stop, enter the same value for arrival_time and departure_time.

To my knowledge you can't have an arrival time but not a departure time, and vice versa, for any stop_time in the trip - they either both need to be populated, or neither should be populated - the exception being the first and last stop_time, which always need both arrival and departure times.

Reply:
Alright, that is not how the python validator is implemented but it makes a lot of sense. Will modify